internal/envoy/auth.go

```diff
@@ -47,13 +47,18 @@ var (
 // UpstreamTLSContext creates an envoy_api_v2_auth.UpstreamTlsContext. By default
 // UpstreamTLSContext returns a HTTP/1.1 TLS enabled context. A list of
 // additional ALPN protocols can be provided.
-func UpstreamTLSContext(ca []byte, subjectName string, alpnProtocols ...string) *envoy_api_v2_auth.UpstreamTlsContext {
+func UpstreamTLSContext(ca []byte, subjectName string, sni string, alpnProtocols ...string) *envoy_api_v2_auth.UpstreamTlsContext {
 	context := &envoy_api_v2_auth.UpstreamTlsContext{
 		CommonTlsContext: &envoy_api_v2_auth.CommonTlsContext{
 			AlpnProtocols: alpnProtocols,
 		},
 	}
 
+	// If SNI is specified, set on the Upstream context
+	if len(sni) > 0 {
+		context.Sni = sni
```

Comment:
You can skip this check; if len(sni) == 0. then context.Sni = sni the same as context.Sni = ""